R/jobUtilities.R

```diff
@@ -316,11 +327,59 @@ waitForTasksToComplete <-
           (taskCounts$validationStatus == "Validated" ||
            totalTasks >= 200000)) {
         cat("\n")
-        return(0)
+        break
       }
 
       Sys.sleep(10)
     }
+
+    cat("Tasks have completed. ")
+    if (enableCloudCombine) {
+      cat("Merging results")
+
+      # Wait for merge task to complete
+      repeat {
+        # Verify that the merge cloud task didn't have any errors
+        mergeTask <- rAzureBatch::getTask(jobId, paste0(jobId, "-merge"))
+
+        # This test needs to go first as Batch service will not return an execution info as null
+        if (is.null(mergeTask$executionInfo$result)) {
+          cat(".")
+          Sys.sleep(5)
+          next
+        }
+
+        if (mergeTask$executionInfo$result == "Success") {
+          break
+        }
+        else {
+          rAzureBatch::terminateJob(jobId)
+
+          # The foreach will not be able to run properly if the merge task fails
+          # Stopping the user from processing a merge task that has failed
+          stop(sprintf(
+            paste(
+              "An error has occurred in the merge task of the job '%s'.",
```

Comment:
We have this error in two places now. Might be worth migrating things like this to their own file and return the text as a function:

```R
# Optional, add params to the method to be more descriptive about the task? e.g. 'merge task' or 'task 4'
getTaskFailedErrorString <- function() {
    text <- "An error has occurred..."
    return text;
}
```